openlibrary/book_providers.py

```diff
@@ -425,8 +425,16 @@ def solr_key(self):
         return None
 
     def get_identifiers(self, ed_or_solr: Edition | dict) -> list[str]:
-        # It's an edition
-        if ed_or_solr.get('providers'):
+        # It's an edition.
+        # `Acquisition` objects have a `provider_name` attribute, and search results are
+        # `Storage` objects with an `providers` being a list of `Acquisition`. Ensure
+        # `provider_name` is `None` so as not to match every web book as a DirectProvider.
+        # See `add_non_solr_fields` in `worksearch/schemes/works.py`.
+        providers = ed_or_solr.get('providers', [])
+        provider_name = (
```

Comment:
This will cause some issues cause provider_name can be set by UI... we should probably not have this be configurable. It should be the trusted book provider `name` if we are generate the acquisition from identifiers, otherwise it should be `direct` to denote a direct book provider.

We drafted a sample to make this more bullet proof.

Also: We'll need to add a `def get_acquisitions` method to the Internet Archive book provider.